File: ccalljava3so/src/main/java/com/atguigu/ccalljava2/MainActivity.java
package com.atguigu.ccalljava2;

import android.app.Activity;
import android.os.Bundle;

public class MainActivity extends Activity {
    private JNI jni;
    {
        System.loadLibrary("ccalljava2");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        jni = new JNI();
        jni.callbackSayHello();
    }
}

File: mykotlintest/src/main/java/com/example/mykotlintest/MainActivity.kt
package com.example.mykotlintest
import android.app.Activity
import android.os.Bundle
class MainActivity : Activity() {
    // c  a  s  k
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
       // Student.printname("nihao")
    }
    override fun onStart() {
        super.onStart()
    }
}
